tests/input/semicolon_spacing.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+declare(strict_types=1);
+
+echo -1    ;
+
+echo 'foo'
+;
```

Comment:
Maybe add an example with something multiline, like a chain of method calls, or maybe a heredoc, because that's the only case where I would consider adding a newline before a semicolon.

Reply:
Maybe something like if that is even supported? :P
````php
$qb->select()
    ->from()
    ->where()
    ;

Reply:
@greg0ire @Majkl578 That should be reported or not, in your opinions?

Reply:
It should be reported and fixed IMO. Just my opinion though.

Reply:
@greg0ire you mean "->where();" fixing or trimming-whitespaces-fixing? Personally I'm leaving `;` in new line for builders and such so adding line doesn't require looking for semicolon and also diff is +1/0 instead of +2/-1 (similar to comma in arrays).

Reply:
I meant "->where();" fixing,  but I never considered the diff issue.

Reply:
What's the correct syntax for multiline calls like the above?
```
$qb
    ->foo()
    ->bar()
;
```

vs.

```
$qb
    ->foo()
    ->bar();

Reply:
I've always used the latter, but have no good argument for it, as opposed to @malarzm who made a good point.

Reply:
Yes, I've started using the former for the very same reason, which is why I'm asking.

Reply:
All in favor of switching to the former, please upvote this. All against, please downvote.

Reply:
@greg0ire You have my upvote, we just need to figure out how to deal with the Sniff. Also, can you please look at the fails? I can’t figure out what is wrong :-1:

Reply:
There must be an extra or missing newline somewhere…

Reply:
Can we please leave it out of this specific patch and use the normal approval (and a dedicated PR) for this?

Reply:
Can you leave the part with the qb out if you have the part with

```
echo 'foo'
;
```

How is it different exactly? If we go with @malarzm proposal then we have an issue don't we?